core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/support/IncludeExclude.java

```diff
@@ -138,6 +169,29 @@ public boolean accept(BytesRef value) {
 
     }
 
+    class PartitionedOrdinalsFilter extends OrdinalsFilter {
+
+
+        @Override
+        public LongBitSet acceptedGlobalOrdinals(RandomAccessOrds globalOrdinals) throws IOException {
+            final long numOrds=globalOrdinals.getValueCount();
+            final LongBitSet acceptedGlobalOrdinals = new LongBitSet(numOrds);
+            final TermsEnum termEnum = globalOrdinals.termsEnum();
+            
+            BytesRef term = termEnum.next();
+            int ord = 0;
+            while (term != null) {
+                int hash = Math.abs(term.hashCode());
+                if (hash % incNumPartitions == incZeroBasedPartition) {
```

Comment:
I would use `Math.floorMod(term.hashCode(), incNumPartitions)` to avoid issues in the case that the hashcode is `Integer.MIN_VALUE`